### PR TITLE
Relax `Arc` cfg; remove `Send` bound from `Valid`; implement `Valid` & `CanonicalDeserialize` for `Rc`

### PR DESCRIPTION
**Pull request by mmagician**
## Description

closes: https://github.com/arkworks-rs/algebra/issues/649

depends on https://github.com/arkworks-rs/std/pull/48

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [x] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

**Comment by weikengchen** [comment]
So, I think the remaining action item is to remove the patch?

**Comment by mmagician** [comment]
@weikengchen no, quite the opposite. It relies on latest `std` master

**Comment by weikengchen** [comment]
Let me revert my change.

**Comment by weikengchen** [comment]
But it looks like ready to merge? I think the changes are easy to follow and seem straightforward.

**Comment by mmagician** [comment]
Yes, it's ready to merge. The dependency PR had already been merged to `std`

**Review by weikengchen — APPROVED** [review]

File: serialize/src/impls.rs [hunk 1]
```diff
@@ -323,7 +349,7 @@ impl<T: CanonicalSerialize + ToOwned> CanonicalSerialize for ark_std::sync::Arc<
     }
 }
 
-#[cfg(feature = "std")]
+#[cfg(target_has_atomic = "ptr")]
 impl<T: Valid + Sync + Send> Valid for ark_std::sync::Arc<T> {
```
Comment (weikengchen) [inline serialize/src/impls.rs:353]: Should this Sync be dropped?

Comment (weikengchen) [reply serialize/src/impls.rs:353]: (Unrelated, closed)

**Pull request merged**